rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -149,8 +149,8 @@ unsafe fn mark_fields<M: Memory>(mem: &mut M, obj: *mut Obj, obj_tag: Tag, heap_
         let field_value = *field_addr;
         mark_object(mem, field_value, heap_base);
 
-        // Thread if backwards pointer
-        if field_value.unskew() < obj as usize {
+        // Thread if backwards or self pointer
+        if field_value.unskew() <= obj as usize {
```

Comment:
Would it make sense to use field_addr to be symmetric with the forward thread in line 214?

Reply:
Good point. I'm not sure if we should make this one like line 214, or make line 214 like this one. Personally I prefer comparing with `obj` instead of field address, as we don't allow internal pointers and I think comparing with `obj` makes this clear. WDYT?

Reply:
Comparing with `obj` sounds good as well as long as we are consistent

Reply:
Even better, a 1-character fix!